pandas/core/arrays/base.py

```diff
@@ -462,22 +463,40 @@ def factorize(self, na_sentinel=-1):
     # ------------------------------------------------------------------------
     # Indexing methods
     # ------------------------------------------------------------------------
-    def take(self, indexer, allow_fill=True, fill_value=None):
-        # type: (Sequence[int], bool, Optional[Any]) -> ExtensionArray
+    def _values_for_take(self):
+        """Values to use for `take`.
+
+        Coerces to object dtype by default.
+
+        Returns
+        -------
+        array-like
+            Must satisify NumPy's `take` semantics.
+        """
+        return self.astype(object)
+
+    def take(self, indexer, fill_value=None, allow_fill=None):
```

Comment:
Switch order of fill_value / allow_fill keywords? (and same comment about `allow_fill` default)

Reply:
Why switch the order?

Reply:
So we do it in this order currently in algos. But, I think I find the reverse more natural, because `fill_value` is dependent on `allow_fill`. 
If I would explain the behaviour of those keywords, I would first say: "depending on allow_fill an indexer of -1 is interpreted as .. or ...", and only then "if allow_fill is True, with fill_value you can specify another value than the default missing value to fill in the non-existing values"

Here I said "allow_fill" correctly :-)

Reply:
I see. I agree with that the other order (`allow_fill`, `fill_value`) is more natural.

I did this order because it matches, but since `take_nd`, aren't in the public API, it doesn't matter so much.